File: ina_2sem/labor_cs/2_lab/cpp/main.cpp
#include <iostream>
#include <algorithm>
#include "ArabRzym.h"
#include "ArabRzymException.h"

using namespace std;

int main(int argc, char* argv[]) {
    for (int i = 1; i < argc; ++i) {
        string arg = argv[i];
        try {
            if (isdigit(arg[0])) {
                int num = stoi(arg);
                string roman = ArabRzym::arab2rzym(num);
                cout << arg << " -> " << roman << endl;
            } else {
                string upperArg = arg;
                transform(upperArg.begin(), upperArg.end(), upperArg.begin(), ::toupper);
                int arabic = ArabRzym::rzym2arab(upperArg);
                cout << arg << " -> " << arabic << endl;
            }
        } catch (const invalid_argument& e) {
            cerr << "Invalid number: " << arg << endl;
        } catch (const ArabRzymException& e) {
            cerr << "Error: " << e.what() << endl;
        }
    }
    return 0;
}

File: ina_2sem/labor_cs/2_lab/java/Main.java
public class Main {
    public static void main(String[] args) {
        for (String arg : args) {
            try {
                if (arg.matches("\\d+")) {
                    int num = Integer.parseInt(arg);
                    String roman = ArabRzym.arab2rzym(num);
                    System.out.println(arg + " -> " + roman);
                } else
                    int arabic = ArabRzym.rzym2arab(arg.toUpperCase());
                    System.out.println(arg + " -> " + arabic);

                }
            } catch (NumberFormatException e) {
                System.err.println("Invalid number: " + arg);
            } catch (ArabRzymException e) {
                System.err.println("Error: " + e.getMessage());
            }
        }
    }
}
